### DISCUSSION TITLE
Importing with json

### DISCUSSION THREAD

**Discussion by Archie-2021** [tridactyl/tridactyl#4925] [Q&A]
Is it possible to import .json config file and set up this plugin with just that?

**Comment by bovine3dom** [comment]
There's `:mktridactylrc --clipboard` and `:source --clipboard`, is there a particular reason you want JSON?

**Comment by Archie-2021** [comment]
> There's `:mktridactylrc --clipboard` and `:source --clipboard`, is there a particular reason you want JSON?

Not really, It is just I am used to do it (with surfingkeys) and this plugin shows the config in json format anyway. 

My main issue is to export and import the settings and I am confused how to do it. What should be the content of clipboard for it to work / How should I get the export from on machine to feed it to another one via clipboard?

**Reply by bovine3dom** [reply]
The content of the clipboard when you run `:source --clipboard` should be the output of `:mktridactylrc --clipboard`, e.g. 

```
bind d composite tabprev; tabclose #
bind D tabclose
```

You can transfer it between machines by e.g. pasting it into an email, sending it and then copying it on the other machine

**Reply by Archie-2021** [reply]
Ok thank you very much. Clipboard works but I cant write that plain text config to a file. The documentation says it should write it relative to .local/share/tridactyl I even cteated the directory manually for it

**Reply by bovine3dom** [reply]
E.g. open up Gedit, paste it in, save it? You can put it wherever you like if you're just going to copy from it later. I'm not sure we understand each other :)

**Comment by Archie-2021** [comment]
> E.g. open up Gedit, paste it in, save it? You can put it wherever you like if you're just going to copy from it later. I'm not sure we understand each other :)

I understand, I know how to copy past stuff, believe me... It is just that when you told the command for it I looked it up and saw there is a write to file option so if it is there why not use it. I am use a git repo to version control my settings, so it is much more convenient for me  if it is possible. If you don't think it is needed I can understand that too. It is your project and your choice, but I like to find how to use write to the file if the options is already there and there is work needed to be done.

Thanks

**Reply by bovine3dom** [reply]
Sorry, I thought you didn't want to use our functionality for that which is why you wanted a JSON dump.

If you install `:native` on both machines, you can use `:mktridactylrc` to write to (usually) `~/.config/tridactyl/tridactylrc` which will then be read at Tridactyl startup. You can use git to control it like any other dotfile.

Just make sure `:native` is installed on each machine

**Comment by Archie-2021** [comment]
> Sorry, I thought you didn't want to use our functionality for that which is why you wanted a JSON dump.
> 
> If you install `:native` on both machines, you can use `:mktridactylrc` to write to (usually) `~/.config/tridactyl/tridactylrc` which will then be read at Tridactyl startup. You can use git to control it like any other dotfile.
> 
> Just make sure `:native` is installed on each machine

No , as a a Linux user I like plaintext configs much better.

It is works now. Thanks
